### PR TITLE
breaking ame fuel makes it leak antimatter (explode)

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
title

it will leak all its fuel over the span of 10 seconds with a special antimatter explosion
antimatter explosion is:
- blue like the jar
- cool
- does 5 extra burn compared to normal explosion

when it explodes it doesn't do much damage but it will be flung around randomly
it can potentially space tiles and destroy things like the ame if it explodes on it enough

## Why / Balance
allows for more syndicate plays and creating some chaos

## Technical details
added leak explosion fields to `AmeFuelContainerComponent`
when it is beaten to 100 damage it will start exploding until there is no more fuel
if fuel is vvd in again it will resume exploding :trollface:

## Media

outdated it solely uses breaking now

https://github.com/space-wizards/space-station-14/assets/39013340/69450526-3465-4933-97dd-27dd14515fa5

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
no

**Changelog**
:cl:
- add: AME fuel jars can now be broken to release antimatter and cause explosions!

### CODE REVIEW

**Comment by mirrorcult** [comment]
id kind of rather this was just a normal thing you could do (no emag) but make the explosions not able to space

**Comment by LankLTE** [comment]
Making station sabotage exclusive to the emag is probably a bad idea so I’d rather not see that be the case.

**Comment by deltanedas** [comment]
> id kind of rather this was just a normal thing you could do (no emag) but make the explosions not able to space

if it was available to selfantags people could just leak all the ame fuel roundstart

**Comment by LankLTE** [comment]
> if it was available to selfantags people could just leak all the ame fuel roundstart
> 
> unless you mean like beating it to 100 damage instead of a verb like gas tank

I'd be something deliberate like exploding the AME currently is so

**Comment by mirrorcult** [comment]
it doesnt need to be as accessible as a verb

**Comment by deltanedas** [comment]
it now breaks above 100 damage
still can be emagged and uses emag logic tho

**Comment by drteaspoon420** [comment]
I still don't see the reason for adding EMAG to the equation. either give it valve like gas canisters or burst on sufficient damage.

**Comment by mirrorcult** [comment]
yeah i dont think emag should be involved in this at all, its extremely unintuitive and its not like AME jars even have any electronics to hack

**Comment by drteaspoon420** [comment]
Well there is such thing as solenoid valve. But those usually have remote control in mind.

**Comment by deltanedas** [comment]
@mirrorcult should i have a bool and special destruction behavior to set it to true, or is there a more suitable thing that already exists

**Comment by deltanedas** [comment]
also fun fact you can chain c4 and antimatter :trollface:

File: Content.Server/Ame/EntitySystems/AmeFuelSystem.cs [hunk 1]
```diff
@@ -27,4 +66,20 @@ private void OnFuelExamined(EntityUid uid, AmeFuelContainerComponent comp, Exami
...
+    public void StartLeaking(EntityUid uid, AmeFuelContainerComponent? comp = null)
+    {
+        if (!Resolve(uid, ref comp) || comp.Leaking)
+            return;
+
+        if (comp.FuelAmount < 1 || !HasComp<ExplosiveComponent>(uid))
+            return;
+
+        // don't instantly explode, give the user a little time to react
+        comp.Leaking = true;
```
Comment (mirrorcult) [inline Content.Server/Ame/EntitySystems/AmeFuelSystem.cs:82]: would rather it added an `AmeFuelLeakingComponent` or something and queried over that so it isnt querying over every jar for no reason

Comment (deltanedas) [reply Content.Server/Ame/EntitySystems/AmeFuelSystem.cs:82]: @mirrorcult while im at to make the destruction behaviour generic, how would i addcomp a component specified in yaml so its not so extremely specific

Comment (drteaspoon420) [reply Content.Server/Ame/EntitySystems/AmeFuelSystem.cs:82]: Some sort of 'fuse' component? Delay and explosion prototype?

Comment (drteaspoon420) [reply Content.Server/Ame/EntitySystems/AmeFuelSystem.cs:82]: or simply re-use some explosive components? from c-4 or explosive pen

**Pull request closed without merging**